x86_symbolic/src/Data/Macaw/X86/Crucible.hs

```diff
@@ -863,31 +877,43 @@ unpack2 ::
       V.Vector n (E sym (BVType c)) ->
       IO a) ->
   IO a
-unpack2 sym e w c v1 v2 k =
+unpack2 bak e w c v1 v2 k =
   divExact w c $ \n ->
-    do v1' <- getBitVal sym v1
-       v2' <- getBitVal sym v2
+    do v1' <- getBitVal bak v1
+       v2' <- getBitVal bak v2
        k (V.fromBV e n c v1') (V.fromBV e n c v2')
 
 -- XXX: Do we want to be strict here (i.e., asserting that the thing is
 -- not a pointer, or should be lenent, i.e., return an undefined value?)
 getBitVal ::
-  IsSymInterface sym =>
-  sym ->
+  (IsSymInterface sym, IsBoolSolver sym bak) =>
+  bak ->
   AtomWrapper (RegEntry sym) (M.BVType w) ->
   IO (E sym (BVType w))
-getBitVal sym (AtomWrapper x) =
-  do v <- projectLLVM_bv sym (regValue x)
+getBitVal bak (AtomWrapper x) =
+  do v <- projectLLVM_bv bak (regValue x)
      case regType x of
        LLVMPointerRepr w -> return (ValBV w v)
        _ -> error "getBitVal: impossible"
 
+getBitVal' ::
```

Comment:
We (for some definition of "we") should leave a comment explaining why both `getBitVal` and `getBitVal'` exist and why you'd want to use the latter over the former in one particular spot. (I know that I'm at fault here for offering a patch that introduced `getBitVal'`, but I'd hate to see this become a permanent fixture without some kind of accompanying documentation.)

Reply:
I was wondering if this should get pulled out into a separate PR that lands before this one.  I think there are still some questions to answer about if this was the right way to fix this.

Reply:
I'll work on a separate PR for this